src/compiler/emitter.ts

```diff
@@ -134,28 +134,30 @@ namespace ts {
             emitDeclarationFileOrBundle(sourceFileOrBundle, declarationFilePath, declarationMapPath);
 
             if (!emitSkipped && emittedFilesList) {
-                if (!emitOnlyDtsFiles) {
-                    emittedFilesList.push(jsFilePath);
-                }
-                if (sourceMapFilePath) {
-                    emittedFilesList.push(sourceMapFilePath);
+                if (!compilerOptions.emitDeclarationOnly) {
+                    if (!emitOnlyDtsFiles) {
+                        emittedFilesList.push(jsFilePath);
+                    }
+                    if (sourceMapFilePath) {
+                        emittedFilesList.push(sourceMapFilePath);
+                    }
+                    if (bundleInfoPath) {
+                        emittedFilesList.push(bundleInfoPath);
+                    }
                 }
                 if (declarationFilePath) {
                     emittedFilesList.push(declarationFilePath);
                 }

```

Comment:
Thank you. This looks good. While you are at it can you please also add 
```typescript
if(declarationMapPath) {
    emittedFilesList.push(declarationMapPath);
}
```